### PR TITLE
add MUXSDKStats updateCustomerData  method

### PR DESCRIPTION
**Pull request by dylanjha**
https://zube.io/muxinc/mux/c/8024

* adds MUXSDKStats.updateCustomerDataForPlayer:withPlayerData:withVideoData

This methods can be used to set or update properties after the SDK has been
initialized

The purpose of exposing this method is that some of our users get the metadata
asynchronously and need to set it after the SDK has been initialized.

----

When ready for release this will be `v1.1.0`

### CODE REVIEW

File: MUXSDKStats/MUXSDKStats/MUXSDKStats.m [hunk 1]
```diff
@@ -222,4 +222,20 @@ + (void)programChangeForPlayer:(nonnull NSString *)name withVideoData:(nullable
     }
 }
 
++ (void)updateCustomerDataForPlayer:(nonnull NSString *)name withPlayerData:(nullable MUXSDKCustomerPlayerData *)playerData withVideoData:(nullable MUXSDKCustomerVideoData *)videoData {
+    MUXSDKPlayerBinding *player = [_viewControllers valueForKey:name];
+    if (player) {
+        if (videoData) {
+            MUXSDKDataEvent *dataEvent = [MUXSDKDataEvent new];
```
Comment (jsanford8) [inline MUXSDKStats/MUXSDKStats/MUXSDKStats.m]: I'd reorganize this a bit, something like:

```objective-c
if (player) {
  MUXSDKDataEvent *dataEvent = [MUXSDKDataEvent new];
  if (videoData) {
    [dataEvent setCustomerVideoData:videoData];
  }
  if (playerData) {
    [dataEvent setCustomerPlayerData:playerData];
  }
  [MUXSDKCore dispatchEvent:dataEvent forPlayer:name];
}
```

Something like that, but you might want to check to see if it was updated at all. I'd just rather not emit two data events if possible.

Comment (dylanjha) [reply MUXSDKStats/MUXSDKStats/MUXSDKStats.m]: Oh ya, good call, much more better

File: Frameworks/iOS/fat/MUXSDKStats.framework/Headers/MUXSDKStats.h [hunk 2]
```diff
@@ -106,4 +106,14 @@ FOUNDATION_EXPORT
...
 
+/*!
+ @method      updateCustomerData:forPlayer:withPlayerData:withVideoData
+ @abstract    allows videoData to be set or updated for the player
+ @param       name The name of the player to update
+ @param       playerData A MUXSDKCustomerPlayerData object with video metadata
+ @param       videoData A MUXSDKCustomerVideoData object with video metadata
+ @discussion  Use this method after you have already initialized the Mux SDK at any time before the player has been destroyed. Pass in either videoData or playerData.
+ */
++ (void)updateCustomerDataForPlayer:(nonnull NSString *)name withPlayerData:(nullable MUXSDKCustomerPlayerData *)playerData withVideoData:(nullable MUXSDKCustomerVideoData *)videoData;
```
Comment (jsanford8) [inline Frameworks/iOS/fat/MUXSDKStats.framework/Headers/MUXSDKStats.h]: Okay, it's possible that I'm remembering style from looong ago, but for some reason I think this should be `updateCustomerPlayerDataForPlayer: withPlayerData: videoData: `.

That _seems_ to be what's in this guide here: https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/CodingGuidelines/Articles/NamingMethods.html. But open to your interpretation, also

File: MUXSDKStats/MUXSDKStats/MUXSDKStats.m [hunk 3]
```diff
@@ -222,4 +222,18 @@ + (void)programChangeForPlayer:(nonnull NSString *)name withVideoData:(nullable
...
+    if (!player) return;
+    if (!playerData && !videoData) return;
+    MUXSDKDataEvent *dataEvent = [MUXSDKDataEvent new];
+    if (playerData) {
+        [dataEvent setCustomerPlayerData:playerData];
+    }
+    if (videoData) {
+        [dataEvent setCustomerVideoData:videoData];
+    }
+    [MUXSDKCore dispatchEvent:dataEvent forPlayer:name];
```
Comment (jsanford8) [inline MUXSDKStats/MUXSDKStats/MUXSDKStats.m]: perfect, looks good!

File: Frameworks/iOS/fat/MUXSDKStats.framework/Headers/MUXSDKStats.h (hunk 2 above)
Comment (dylanjha) [reply Frameworks/iOS/fat/MUXSDKStats.framework/Headers/MUXSDKStats.h]: Discussed IRL. We can't decide what the clear convention is so we're going with what we have

**Pull request merged**